build/rewrite.go

```diff
@@ -54,17 +57,73 @@ func allowedSort(name string) bool {
 	return false
 }
 
-// Rewrite applies the high-level Buildifier rewrites to f, modifying it in place.
+// Rewrites controls the rewrites to be applied.
+//
+// If non-nil, the rewrites with the specified names will be run. If
+// nil, a default set of rewrites will be used that is determined by
+// the type (BUILD vs default starlark) of the file being rewritten.
+type Rewriter struct {
+	DisableRewrites                 *[]string
+	LabelArg                        *map[string]bool
+	Denylist                        *map[string]bool
+	SortableListArg                 *map[string]bool
+	SortDenyList                    *map[string]bool
+	SortAllowList                   *map[string]bool
+	NamePriority                    *map[string]int
+	StripLabelLeadingSlashes        *bool
+	ShortenAbsoluteLabelsToRelative *bool
+}
+
 func Rewrite(f *File) {
+	var rewriter = &Rewriter{
```

Comment:
This can now just be:
```
var rewriter := &Rewriter{}
rewriter.Rewrite(f)
```

Reply:
I think the code would be a lot clearer if the Rewriter didn't have pointers for all of its members and the Rewrite function initialized the Rewriter with the values from the tables. Then the rest of the code can just use the value from the Rewriter instance directly.

Reply:
We ended up going with this decision because we did not want to modify the Rewrite object when its passed through

Reply:
This would not require modifying any objects that are passed in. This just would change it so that when the Rewriter object is created in the Rewrite function it gets initialized with the tables values. The rest of the code doesn't need to modify anything, it can just rely on the values in the Rewriter instance being authoritative.